acmewrapper.go

```diff
@@ -112,6 +112,10 @@ func (w *AcmeWrapper) TLSConfigGetCertificate(clientHello *tls.ClientHelloInfo)
 // necessary.
 func (w *AcmeWrapper) TLSConfig() *tls.Config {
 	return &tls.Config{
+		//Go 1.6 "allows Listen to succeed when the Config has a nil Certificates, as long as the
+		//GetCertificate callback is set" See https://golang.org/doc/go1.6#minor_library_changes.
+		//So to add 1.5 support, we provide the default certificate in addition to GetCertificate.
+		Certificates:   []tls.Certificate{*w.cert},
```

Comment:
This unfortunately won't work:

Renewing certificates won't reflect the change. After renew, the certificate in acmewrapper will change, but having this will make the server still serve the old cert.

One option would be to manually change this array each time renew happens, but this is not threadsafe. If we happened to change the cert at the moment when it is requested, our server could panic.

I am not sure if the tls configuration is used internally in go vs the info taken into the server and processed. If it is processed, then there is no hope for go1.5 support. Whatever the case, it would require tests to make sure it behaves as expected.

As for adding "hacky" support for go 1.5 if the server uses the config internally, I see a couple options:
- Notifying how to do it in the docs, with the caveat that it could cause a panic, and letting people do it at their own risk
- Adding support automatically by checking if version is 1.5 and logging a warning.

What do you think?